modules/phase_field/src/ics/SpecifiedSmoothSuperellipsoidIC.C

```diff
@@ -0,0 +1,73 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SpecifiedSmoothSuperellipsoidIC.h"
+#include "MooseRandom.h"
+
+template<>
+InputParameters validParams<SpecifiedSmoothSuperellipsoidIC>()
+{
+  InputParameters params = validParams<SmoothSuperellipsoidBaseIC>();
+  params.addClassDescription("Multiple smooth superellipsoids with manually specified center points; semiaxes a,b,c; and exponents n");
+  params.addRequiredParam<std::vector<Real> >("x_positions", "The x-coordinate for each superellipsoid center");
+  params.addRequiredParam<std::vector<Real> >("y_positions", "The y-coordinate for each superellipsoid center");
+  params.addRequiredParam<std::vector<Real> >("z_positions", "The z-coordinate for each superellipsoid center");
```

Comment:
No, we are already using a delimiter for std::vector<std::vector<T> >. I
don't think we really want to introduce a second delimiter.

On Thu, Apr 7, 2016 at 8:16 AM Daniel Schwen <notifications@github.com>
wrote:

> In modules/phase_field/src/ics/SpecifiedSmoothSuperellipsoidIC.C
> <https://github.com/idaholab/moose/pull/6712#discussion_r58878312>:
>
> > +/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
> > +/*                                                              */
> > +/*          All contents are licensed under LGPL V2.1           */
> > +/*             See LICENSE for full restrictions                */
> > +/****************************************************************/
> > +#include "SpecifiedSmoothSuperellipsoidIC.h"
> > +#include "MooseRandom.h"
> > +
> > +template<>
> > +InputParameters validParams<SpecifiedSmoothSuperellipsoidIC>()
> > +{
> > +  InputParameters params = validParams<SmoothSuperellipsoidBaseIC>();
> > +  params.addClassDescription("Multiple smooth superellipsoids with manually specified center points; semiaxes a,b,c; and exponents n");
> > +  params.addRequiredParam<std::vector<Real> >("x_positions", "The x-coordinate for each superellipsoid center");
> > +  params.addRequiredParam<std::vector<Real> >("y_positions", "The y-coordinate for each superellipsoid center");
> > +  params.addRequiredParam<std::vector<Real> >("z_positions", "The z-coordinate for each superellipsoid center");
>
> I wonder if moving this to the new semicolon separated syntax would make
> sense?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6712/files/1df8c332b7ee0f386fcab5a45def164be2dc4ecd#r58878312>
>